### ISSUE TITLE
App restarted when hot reload 

### ISSUE THREAD

**Issue by AliDeeb** [kherel/animated_theme_switcher#51]

**Comment by AliDeeb** [comment]
Because you are defined ThemeModel inside build method in ThemeProviderState class and in every build generate and new key.
please define ThemeModel as a state to fixed it.

**Comment by rishabh-os** [comment]
I have a related issue, which I managed to fix but I am wondering how exactly this occurs and if there is a workaround for this.
If I define my themes in a different file as follows:

```
ThemeData light = ThemeData.light();
ThemeData dark = ThemeData.dark();
```
and pass them to `initTheme` as `initTheme = dark`  the app retains state after a hot reload.

However, if I define the themes in a class, as follows:

```
class Themes {
  static ThemeData light = ThemeData.light();
  static ThemeData dark = ThemeData.dark();
}
```
and then pass it to `initTheme` as `initTheme = Themes.dark`, the app loses state and sends me back to the home screen.

**Comment by kherel** [comment]
hopefully fixed in 2.0.8
